email/src/main/java/edu/jhuapl/dorset/demos/EmailManager.java

```diff
@@ -0,0 +1,308 @@
+    * 
+    * @param config  Configuration object that stores mail server information, username, and password
+    * @throws MessagingException   if connection cannot be established
+    */
+    public EmailManager(Config config) throws MessagingException {
+        
+        user = config.getString(USERNAME_KEY);
+        password = config.getString(PASSWORD_KEY);
+        mailStoreType = config.getString(MAIL_STORE_TYPE_KEY);
+        host = config.getString(HOST_KEY); 
+        from = config.getString(FROM_KEY);
+        Properties prop = extractProperties(config);
+        session = Session.getDefaultInstance(prop);
+        store = session.getStore(mailStoreType);
+        try {
+            store.connect(host, user, password);
+        } catch (MessagingException e) {
+            throw new MessagingException("Failed to set up imap connection", e);
+        }
+        try {
+            initFolders();
+            inboxFolder.open(Folder.READ_WRITE);
+        } catch (MessagingException e) {
+            throw new MessagingException("Failed to initialize and open folder");
+        }
+    }
+    
+    /**
+     * Extract properties from configuration object
+     * 
+     * @param config  Configuration object that stores mail server information, username, and password
+     * @return the properties file
+     */
+    private Properties extractProperties(Config config) {
+        Properties prop = new Properties();
+        for (java.util.Map.Entry<java.lang.String, ConfigValue> entry : config.entrySet()) {
+            prop.setProperty(entry.getKey(), config.getString(entry.getKey()));
+        }        
+        return prop;
+    }
+    /**
+     * Get folders and create them if they do not already exist
+     * @throws MessagingException   if folders cannot be initialized
+     */
+    private void initFolders() throws MessagingException {
+        if (!store.getFolder(FolderType.ERROR.getValue()).exists()) {
+            store.getFolder(FolderType.ERROR.getValue()).create(Folder.HOLDS_MESSAGES);
+        }
+        if (!store.getFolder(FolderType.COMPLETE.getValue()).exists()) {
+            store.getFolder(FolderType.COMPLETE.getValue()).create(Folder.HOLDS_MESSAGES);
+        }
+        inboxFolder = store.getFolder(FolderType.INBOX.getValue());
+        errorFolder = store.getFolder(FolderType.ERROR.getValue());
+        completeFolder = store.getFolder(FolderType.COMPLETE.getValue());
+    }
+
+    /**
+     * Get folder
+     * 
+     * @param folder   the folder to be returned
+     * @return the folder to be accessed
+     */
+    private Folder getFolder(FolderType folder) {
+        switch (folder) {
+            case INBOX:
+                return inboxFolder;
+            case ERROR:
+                return errorFolder;
+            case COMPLETE:
+                return completeFolder;
+            default:
+                //TODO throw an error here instead of returning null
```

Comment:
Since we're using an Enum, there are only three possible states for folder so I don't think you need a default here.